File: src/algorithm_java/leetcode/Problem169.java
package algorithm.leetcode;

public class Problem169 {

    public int majorityElement(int[] nums) {
        int count = 0;
        int ele = -1;
        for (int i = 0; i < nums.length; i++) {
            if (count == 0) {
                ele = nums[i];
                count = 1;
            } else {
                if (nums[i] == ele) {
                    count++;
                } else {
                    count--;
                }
            }
        }
        return ele;
    }
}

File: src/algorithm_cpp/problem169.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
	int count = 0;
	int majority = 0;
	for (int i = 0; i < nums.size(); i++) {
	    if (count == 0) majority = nums[i];
	    if (majority == nums[i]) {
		count++;
	    } else {
		count--;
	    }
	}
	return majority;
    }
};
